echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/DockerEventMonitor.java

```diff
@@ -103,12 +103,19 @@ private boolean matchTags(String suppliedTag, String incomingTag) {
     DockerEvent dockerEvent = (DockerEvent) event;
     String account = dockerEvent.getContent().getAccount();
     String repository = dockerEvent.getContent().getRepository();
-    String tag = dockerEvent.getContent().getTag();
     return trigger -> trigger.getType().equals(TRIGGER_TYPE) &&
-      trigger.getRepository().equals(repository) &&
-      trigger.getAccount().equals(account) &&
-      ((trigger.getTag() == null && !tag.equals("latest"))
-        || trigger.getTag() != null && matchTags(trigger.getTag(), tag));
+              trigger.getRepository().equals(repository) &&
```

Comment:
should just be `return isMatchingTrigger(trigger)` and put all the logic in that method

Reply:
I just pushed a commit that I think does what you're asking for. Had to drag along the DockerEvent also, though.